pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -875,6 +875,18 @@ type PrometheusSpec struct {
 	// Maximum number of bytes used by the Prometheus data.
 	RetentionSize ByteSize `json:"retentionSize,omitempty"`
 
+	// ShardRetentionPolicy defines the retention policy for the Prometheus shards.
+	// (Alpha) Using this field requires the ShardRetentionPolicies feature gate to be enabled.
+	//
+	// The final goals for this feature can be seen at https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202310-shard-autoscaling.md#graceful-scale-down-of-prometheus-servers,
+	// however, the feature is not yet fully implemented and has known limitations:
+	// * Scale up after scale down will not re-use the previously marked-for-deletion shards.
```

Comment:
In the proposal, we say that, in case a scale up happens after a scale down that left shards undelete, we would "turn on" the previous shard instead of creating a third one.

See: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202310-shard-autoscaling.md#scaling-up-after-scaling-down

Reply:
Right and it wouldn't be the case with your PR?

Reply:
Hmmm, I haven't tested that actually 🤔 

I haven't done anything specifically for that to work, so I assumed it just wasn't working like that. I'll try to test it out later today and see what happens